packages/analytics-core/src/plugins/destination.ts

```diff
@@ -79,15 +84,33 @@ export class Destination implements DestinationPlugin {
     });
   }
 
-  addToQueue(...list: Context[]) {
-    const tryable = list.filter((context) => {
+  getTryableList(list: Context[]) {
+    return list.filter((context) => {
       if (context.attempts < this.config.flushMaxRetries) {
         context.attempts += 1;
         return true;
       }
       void this.fulfillRequest([context], 500, MAX_RETRIES_EXCEEDED_MESSAGE);
       return false;
     });
+  }
+
+  scheduleTryable(list: Context[]) {
```

Comment:
Can we reuse this function in `addToQueue()` for cleanliness?

Reply:
Those are different because in addToQueue we want to contact the queue, but for other retry we don't. 
But since most of the code is the same added a flag to enable/disable concat the queue. Please check later comment.

Reply:
Update in https://github.com/amplitude/Amplitude-TypeScript/pull/689/commits/e2151480f4e997ce9dd3b65023cf58b8c29e203a

Reply:
Update in https://github.com/amplitude/Amplitude-TypeScript/pull/689/commits/e2151480f4e997ce9dd3b65023cf58b8c29e203a